src/openms/source/ANALYSIS/QUANTITATION/AbsoluteQuantitation.cpp

```diff
@@ -0,0 +1,438 @@
+    const String & feature_name,
+    const String & transformation_model,
+    const Param & transformation_model_params)
+  {
+    // extract out the calibration points
+    TransformationModel::DataPoints data;
+    TransformationModel::DataPoint point;
+    for (size_t i = 0; i < component_concentrations.size(); i++){
+      point.first = component_concentrations[i].actual_concentration;
+      point.second = component_concentrations[i].feature.getMetaValue(feature_name);
+      data.push_back(point);
+    }
+
+    // fit the data to the model
+    AbsoluteQuantitationMethod aqm;
+    Param params = aqm.fitTransformationModel(transformation_model, data, transformation_model_params);
+
+    // store the information about the fit
+    return params;
+  }
+  
+  double AbsoluteQuantitation::applyCalibration(const Feature & component,
+    const Feature & IS_component,
+    const String & feature_name,
+    const String & transformation_model,
+    const Param & transformation_model_params)
+  {
+    // calculate the ratio
+    double ratio = calculateRatio(component, IS_component, feature_name);
+
+    // calculate the absolute concentration
+    double calculated_concentration = 0.0;
+    AbsoluteQuantitationMethod aqm;
+    calculated_concentration = aqm.evaluateTransformationModel(
+      transformation_model, ratio, transformation_model_params);
+
+    // check for less than zero
+    if (calculated_concentration < 0.0)
+    {
+      calculated_concentration = 0.0;
+    }
+
+    return calculated_concentration;
+  }
+
+  void AbsoluteQuantitation::quantifyComponents(FeatureMap& unknowns)
+  {
+    //Potential Optimizations: create a map for each unknown FeatureMap
+    // to reduce multiple loops
+
+    // initialize all quant_method variables
+    std::map<String,AbsoluteQuantitationMethod>::iterator quant_methods_it;
+    String quant_component_name; //i.e., quant_method transition_id
+    String quant_IS_component_name; //i.e., quant_method internal standard transition_id
+    String quant_feature_name; //i.e., quant_method peak_apex_int or peak_area
+    String transformation_model;
+    Param transformation_model_params;
+
+    // initialize all unknown variables
+    String component_name; //i.e., transition_id
+    String IS_component_name; //i.e., internal standard transition_id
+    String component_group_name; //i.e., peptideRef
+    double calculated_concentration;
+    String concentration_units;
+
+    // initalize all other variables
+    bool IS_found;
+    Feature empty_feature;
+    size_t IS_component_it, IS_component_group_it;
+
+    // // iterate through the unknowns
+    // for (size_t i = 0; i < unknowns.size(); i++)
+    // {      
+
+    // iterate through each component_group/feature     
+    for (size_t feature_it = 0; feature_it < unknowns.size(); ++feature_it)
+    {
+      component_group_name = (String)unknowns[feature_it].getMetaValue("PeptideRef");
```

Comment:
@timosachsenberg @dmccloskey we should find a better way to store this information than in metavalues, e.g. we probably need to change the `Feature` class at some point. E.g. each feature should be able to store a unique reference to its peptide...